### PR TITLE
Add fragment order tests according to the servlet specification

### PR DESCRIPTION
**Pull request by tbitonti**
section "8.2.2. Ordering of web.xml and web-fragment.xml".

Fix NPE which occurs when an unnamed fragment contains an others relative order element.

### CODE REVIEW

**Comment by tbitonti** [comment]
#libby

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
#build

File: dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java [hunk 1]
```diff
@@ -0,0 +1,732 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
```
Comment (pmd1nh) [inline dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: update to 2023

Comment (pmd1nh) [reply dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: also update dev/com.ibm.ws.container.service/src/com/ibm/ws/container/service/config/internal/WebFragmentsInfoImpl.java copyright to 2023

File: dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java [hunk 2]
```diff
@@ -0,0 +1,732 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
```
Comment (tevans78) [inline dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: Please ensure that you use the updated EPL 2.0 copyright header for all OL files.

File: dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java (hunk 1 above)
Comment (tbitonti) [reply dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: Updated.

File: dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java (hunk 2 above)
Comment (tbitonti) [reply dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: Updated.

**Comment by tbitonti** [comment]
#libby

**Comment by tbitonti** [comment]
Single unrelated failure:

no component | 91200 | 0 | 1 |   |  
com.ibm.ws.ui.tool.javaBatch_fat.gui FAT tests | 61 | 0 | 1 |   |  
com.ibm.ws.ui.tool.javaBatch_fat.gui.tests.JavaBatchBasicTests | 9 | 0 | 1 | 10 m 29 s | 3
testJobInstanceGridActionsForReader_EE9_FEATURES

**Comment by tbitonti** [comment]
The text "master" appears in link text to product specifications outside of the Open Liberty managed repository.  The link is correct and cannot be changed.

**Comment by tbitonti** [comment]
*Test Note*

This update was intended as a test only update.  On review of the Servlet specification for Jakarta EE 10, several ordering examples were noted which are idea unit tests.

Five new unit tests were added.  The new tests are based on examples in section 8.2.2. "Ordering of web.xml and web-fragment.xml", of the servlet specification.

Tests were added as "dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java"

* FragmentOrderingTest.java
  * void testFragmentOrderingExample0a()
  * void testFragmentOrderingExample0b()
  * void testFragmentOrderingExample1()
  * void testFragmentOrderingExample2()
  * void testFragmentOrderingExample3()

The test methods correspond with the examples provided by the specification.  The "Example 0" methods correspond with unnumbered examples which are provided directly in the specification text.  "Example1" through "Example3" appear as explicit examples.

---

This update includes a fix for an NPE which was discovered while running the unit tests.  The NPE occurs in the rare circumstance in which relative ordering information is provided by an unnamed web fragment.

**Comment by tbitonti** [comment]
#build

**Review by pmd1nh — DISMISSED** [review]

**Comment by tbitonti** [comment]
#libby

File: dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java [hunk 3]
```diff
@@ -0,0 +1,731 @@
...
+import com.ibm.ws.javaee.dd.web.WebApp;
+import com.ibm.ws.javaee.dd.web.WebFragment;
+import com.ibm.ws.javaee.dd.web.common.AbsoluteOrdering;
+import com.ibm.ws.javaee.dd.web.common.Ordering;
+import com.ibm.wsspi.adaptable.module.Container;
+import com.ibm.wsspi.adaptable.module.UnableToAdaptException;
+
+public class FragmentOrderingTest {
+
+// https://github.com/jakartaee/servlet/blob/master/spec/src/main/asciidoc/servlet-spec-body.adoc
```
Comment (tevans78) [inline dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: better to refer to a specific version of the servlet spec ... which will have the added benefit of avoiding the prohibited word!!

Comment (tbitonti) [reply dev/com.ibm.ws.container.service/test/com/ibm/ws/container/service/config/internal/FragmentOrderingTest.java]: Changed to the "6.0" branch.

**Comment by tbitonti** [comment]
#libby

**Review by tevans78 — APPROVED** [review]
Approved the change to the Servlet spec URL ... I've not checked any of the other code.

**Pull request merged**